### ISSUE TITLE
Auto detect to use polkitd.service

### ISSUE THREAD

**Issue by VorpalBlade** [FedeDP/Clightd#16]
After installing this on Ubuntu 16.04 I'm getting the following error:
```
$ sudo systemctl start clightd
Failed to start clightd.service: Unit polkit.service not found.
```
Reading the file it appears that you indeed knew about this (there is a comment about Ubuntu 16.04), but it would be nice to auto-detect it and fix it at compile or install time.

**Comment by VorpalBlade** [comment]
This would be doable with cmake, but somewhat clunky. I don't think there is any built in support for this, instead requiring rolling our own code.

**Comment by FedeDP** [comment]
> This would be doable with cmake, but somewhat clunky. I don't think there is any built in support for this, instead requiring rolling our own code.

Yes, i think the best is to use lsb-release, as per this answer in SO: https://stackoverflow.com/a/48080035/6065692

I mean, to add lsb_release as a compile-time needed program too, obviously. Or just parsing /etc/lsb-release ourselves...

**Comment by VorpalBlade** [comment]
I'm not sure that is the best approach. I have not looked into the issue, but aren't there other distros also using the polkitd name for the service? I would more expect it to be tied to the version of polkit than to the distro (but I could be wrong). The most reliable solution would seem to be to interrogate the system using systemctl to find out which one is installed.

**Comment by FedeDP** [comment]
Yes this is better still!  
Something like `systemctl list-units | grep polkit | awk '{print $1}'` should do the trick, i guess.

**Comment by FedeDP** [comment]
What about using "find_file" to look for polkitd.service in /lib/systemd/system/, as per: https://packages.ubuntu.com/xenial/amd64/policykit-1/filelist ? 
And then configuring clightd.service to use this variable.

**Comment by VorpalBlade** [comment]
That might work, I'm slightly worried about what happens when building in build roots (could be solved by ensuring that policykit-1 is a build dependency), as well as what happens when using distros with a non-standard file system configuration, such as NixOS.

On the other hand, find_file is definitely better than invoking systemctl when cross compiling.

**Comment by FedeDP** [comment]
This should be fixed in 3.0. It is not a very smart check (ie: if it finds "polkitd.service" where ubuntu installs it, it uses "polkitd" else it uses "polkit"), but it should to the trick.

I'm not willing to spend much time on a 2.5yrs old ubuntu release :)
